src/test/java/org/GehtSoftHWByAziz/HW1/CustomListTest.java

```diff
@@ -0,0 +1,292 @@
+    for (int i = 0; i < 100; i++) {
+      myList.add(i);
+    }
+    assertTrue(myList.containsAll(temp));
+  }
+
+  @Test
+  void testAddAll() {
+
+    var temp = new ArrayList<>(List.of(0,1,2));
+    myList.addAll(temp);
+    for (int i = 0 ; i < 3; i ++) {
+      assertEquals(i, myList.get(i));
+    }
+
+  }
+
+  @Test
+  void removeAll() {
+
+    var temp = new ArrayList<>(List.of(0,1,2));
+    myList.addAll(List.of(0,1,2));
+    myList.removeAll(temp);
+    assertEquals(0,myList.size());
+    assertThrows(RuntimeException.class, () -> myList.get(0));
+  }
+
+  @Test
+  void testIndexOf() {
+    myList.add(100);
+    assertEquals(0, myList.indexOf(100));
+  }
+  @Test
+  void testLastIndexOf() {
+    myList.add(100);
+    myList.add(100);
+    myList.add(100);
+    myList.add(100);
+
+    assertEquals(3, myList.lastIndexOf(100));
+  }
+  @Test
+  void testIteratorAdd() {
+    var it = myList.iterator();
+
+    for (int i = 0; i < 100; i++) {
+      myList.add(i);
+    }
+    int i = 0;
+    while (it.hasNext()) {
+      var element = it.next();
+      assertEquals(myList.get(i), element);
+      i++;
+    }
+    assertEquals(100, i);
+  }
+
+  @Test
+  void testListIteratorAdd() {
+    var it = myList.listIterator();
+
+    for (int i = 0; i < 100; i++) {
+      myList.add(i);
+    }
+    int i = 0;
+    while (it.hasNext()) {
+      var element = it.next();
+      assertEquals(myList.get(i), element);
+      i++;
+    }
+    assertEquals(100, i);
+    while (it.hasPrevious()) {
+      i--;
+      var element = it.previous();
+      assertEquals(element, myList.get(i));
+    }
+    assertEquals(0, i);
+  }
+
+  @Test
+  void testIteratorRemove() {
+    var it = myList.listIterator(100);
+    for (int i = 0; i < 100; i++) {
+      myList.add(i);
+    }
+    int i = 100;
+    while (it.hasPrevious()) {
+      it.previous();
+      it.remove();
+    }
+    assertEquals(0, myList.size());
+  }
+
+  @Test
+  void testIteratorSet() {
+    var it = myList.iterator();
+    for (int i = 0; i < 100; i++) {
+      myList.add(i); // to increase size to 100
+    }
+    for (int i = 0; i < 100; i++) {
+      myList.set(i, 11);
+    }
+    for (var i : myList) {
+      assertEquals(11, i);
+    }
+    assertEquals(100, myList.size());
+  }
+
+  @Test
+  void testArrayListAndMyListAndCheckSublist() {
+    for (int i = 0; i < 100; i++) {
+      myList.add(i);
+      arrList.add(i);
+    }
+    assertEquals(myList.size(), arrList.size());
+
+    for (int i = 0; i < 100; i++) {
+    assertEquals(myList.get(i), arrList.get(i));
+    }
+    var temp  = myList.subList(0,100);
+    myList.removeAll(arrList);
+    arrList.removeAll(temp);
+
+    assertEquals(myList.size(),arrList.size());
+    assertEquals(0, myList.size());
+  }
+
+  @Test
+  void testEqualsAndHash() {
+
+    for (int i = 0; i < 100; i++) {
+      myList.add(i);
+      arrList.add(i);
+    }
+    assertEquals(myList, arrList);
+    assertEquals(myList.hashCode(),arrList.hashCode());
+  }
+}
```

Comment:
try to adjust the test using @ParametrizedTest annotation to avoid code duplication